spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraBatchOperations.java

```diff
@@ -69,6 +69,16 @@
 	CassandraBatchOperations insert(Iterable<?> entities);
 
 	/**
+	 * Add a collection of inserts with a given {@code ttl} to the batch.
+	 *
+	 * @param entities the entities to insert; must not be {@literal null}.
+	 * @param ttl the TTL to apply.
+	 * @return {@code this} {@link CassandraBatchOperations}.
+	 * @throws IllegalStateException if the batch was already executed.
+	 */
+	CassandraBatchOperations insert(Iterable<?> entities, int ttl);
```

Comment:
Let's use `WriteOptions` in the signature instead of ttl. `INSERT` statements support much more options on statement-level.